staging/src/k8s.io/client-go/tools/cache/mutation_detector.go

```diff
@@ -118,6 +118,7 @@ func (d *defaultCacheMutationDetector) CompareObjects() {
 		if !reflect.DeepEqual(obj.cached, obj.copied) {
 			fmt.Printf("CACHE %s[%d] ALTERED!\n%v\n", d.name, i, diff.ObjectGoPrintSideBySide(obj.cached, obj.copied))
 			altered = true
+			break
```

Comment:
Here's some general guidelines:
* Understand why a thing is there before you remove it ("chesterton's fence")
* Understand if a thing is important, and by what measure
* Send a test or a benchmark demonstrating the change is safe and/or important

In this case, the point of the loop is to run the Printf on EVERY affected object.

Reply:
@lavalamp 
My original thinking was that the traversal may take some time beyond the detection of the first altered entry.
Now I think more and agree that listing all altered entries would facilitate troubleshooting.

Thanks for taking a look.

Reply:
@lavalamp 
I have a question after reading AddObject:
```
		d.cachedObjs = append(d.cachedObjs, cacheObj{cached: obj, copied: copiedObj})
```
the cached and copied are kept in the same cacheObj.
I wonder if there is a chance that the cached and copied are altered in such a way that DeepEqual wouldn't detect change between the two - though cached was indeed altered.